handling-forms/controlled-forms.md

```diff
@@ -637,24 +713,126 @@ export default NewStudentForm;
 
 ### !end-callout
 
+### !callout-info
+
+## Refactoring Opportunity: Combining Multiple Change Handlers into One
+
+Notice that the functions `onNameChange` and `onEmailChange` are nearly identical. The only difference between the two event handlers is that one updates the value for the name property and one updates the value for the email property in the formFields state object.
+
+<br/>
+
+We can refactor our code to combine  `onNameChange` and `onEmailChange` into one event handler called `handleChange` to reduce repetition in the `NewStudentForm` component.
+
+<br/>
+
+We can accomplish this by using the spread operator notation with object shorthand notation as we saw in the lesson on "Passing Down Event Handlers".
+
+```js
+import { useState } from 'react';
+
+const NewStudentForm = () => {
+
+    const [formFields, setFormFields] = useState({
+        name: '',
+        email: ''
+    });
```

Comment:
`fullName`/`name` is a problem here. We need to mention that the `input` `name` needs to match the state key name so that we can use the `name` to figure out the key. Since the `name` key is what's used in a student record, we'd probably change the `name` of the input to match, which gives us the potentially confusing `name="name"`, which is I'm sure why the original example went with `fullName` instead. But we should at least mention all this.